lib/clamp-device.in

```diff
@@ -197,7 +203,7 @@ fi
 
 # Optimization notes:
 #  -disable-simplify-libcalls:  prevents transforming loops into library calls such as memset, memcopy on GPU
-$OPT -mtriple amdgcn--amdhsa-amdgiz -mcpu=$AMDGPU_TARGET -infer-address-spaces -amdgpu-internalize-symbols -disable-simplify-libcalls $KMOPTOPT -verify $2.selected.bc -o $2.opt.bc
+$OPT -mtriple amdgcn--amdhsa-amdgiz -mcpu=$AMDGPU_TARGET -amdgpu-internalize-symbols -disable-simplify-libcalls $KMOPTOPT -verify $2.selected.bc -o $2.opt.bc
```

Comment:
@AlexVlx so it seems @scchan is proposing to add `-promote-pointer-kernargs-to-global -infer-address-spaces` here, perhaps after `$KMOPTOPT (-O3)` and before `-verify` so the passes be inserted *after* `-inline`?

Reply:
Definitely not after running optimisation, that will prune the casts. The initial ordering (infer before inline) was a typo on my part, I only realised what you guys meant now; corrected.